app/src/main/java/org/mozilla/focus/activity/MainMediator.java

```diff
@@ -63,6 +63,14 @@ public void showUrlInput(@Nullable String url) {
     }
 
     public void showBrowserScreen(@NonNull String url, boolean clearHistory) {
+        clearInputAndCommitBrowserFragment(prepareBrowsing(url, clearHistory));
+    }
+
+    public void showBrowserScreenForRestoreTabs(@NonNull String tabId) {
+        clearInputAndCommitBrowserFragment(prepareBrowsingForRestoreTabs(tabId));
+    }
+
+    private void clearInputAndCommitBrowserFragment(FragmentTransaction browserFragmentTransaction) {
```

Comment:
I feel like intention of this method is not clear. Actually its implementation is to 1) clear input fragment 2) commit a transaction.

Here comes a question: what will it happen if we `browserFragmentTransaction.commit()` ? We have no clue about it. I suggest to change this method to `clearInputFragmentImmediate()`, then use this method in `showBrowserScreen` and `showBrowserScreenForRestoreTabs` respectively.